src/shared/release_info.py

```diff
@@ -82,3 +85,66 @@ def get_revision_to_track(all_revisions_tags: list) -> dict:
 
         revision_track[revision] = track
     return revision_track
+
+
+def get_revision_to_released_tags(all_releases: dict) -> dict:
+    """
+    Iterates over the provided dictionary with all the releases
+    and extracts the revision numbers and their corresponding
+    released tags. The resulting dictionary maps each revision
+    number to a list of released tags.
+    """
+    revision_to_released_tags = defaultdict(list)
+    tag_mapping_from_all_releases = get_tag_mapping_from_all_releases(all_releases)
+    for tag, revision in tag_mapping_from_all_releases.items():
+        if not revision.isdigit():
+            visited = set()
+
+            def find_alias_revision(rev: str, visited: set) -> str:
+                if rev in visited:
+                    raise BadChannel(
+                        f"Tag {tag} was caught in a circular dependency, "
+                        "following tags that follow themselves. Cannot pin a revision."
+                    )
+                visited.add(rev)
+                if not rev.isdigit():
+                    return find_alias_revision(
+                        tag_mapping_from_all_releases[rev], visited
+                    )
+                return rev
+
+            revision = find_alias_revision(revision, visited)
+        revision = int(revision)
+        revision_to_released_tags[revision].append(tag)
+
+    for revision, tags in revision_to_released_tags.items():
+        revision_to_released_tags[revision] = sorted(tags)
+
+    return dict(revision_to_released_tags)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
```

Comment:
I think we should create an individual store_true flag to execute get_revision_to_released_tags. We could even make it mutually exclusive of other "command" flags if required. That way it is integrated into the --help info.